File: C++/Chapter 1/4.cpp
#include <iostream>

bool is_prime(unsigned int number){
    if (number <= 3){
        return number>1? true: false;
    }
    
    int range_number = [];

    return true;
}

void show_prime_number_lower_than(unsigned int number){
    for(unsigned int i=number-1; 1; i--){
        if (is_prime(i)){
            std::cout << "The greatest prime number lower than input = " << i << std::endl;
            break;
        }
        
    }

}

int main(){
    std::cout << "The greates prime number lowest than input value" << std::endl;
    std::cout << "Input number: " << std::endl;
    double number;
    std::cin >> number;
    show_prime_number_lower_than(number);
}

File: Python/Chapter 1/4.py
def is_prime(number):
    if number <= 3:
        return number > 1

    range_numbers = range(2, round(number / 2))

    for num in range_numbers:
        if number % num == 0:
            return False
    return True
    
if __name__ == '__main__':
    print ('The greates prime number lower than input value')
    print ('Input a: ', end='')
    a = eval(input())

    for i in range(a-1, 1, -1):
        if is_prime(i):
            print (f'The greates prime number lower than input value = {i}')
            break